x-pack/plugins/security/server/routes/api_keys/create.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import type { RouteDefinitionParams } from '..';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+export function defineCreateApiKeyRoutes({
+  router,
+  getAuthenticationService,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/api_key',
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          expiration: schema.maybe(schema.string()),
+          role_descriptors: schema.recordOf(
+            schema.string(),
+            schema.object({}, { unknowns: 'allow' }),
+            {
+              defaultValue: {},
+            }
+          ),
+        }),
+      },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      try {
+        const apiKey = await getAuthenticationService().apiKeys.create(request, request.body);
+
+        if (!apiKey) {
+          return response.badRequest({ body: { message: `API Keys are not available` } });
+        }
```

Comment:
It was not clear to me at first that the only way you're going to get a null / "falsey" result is if API keys are not enabled.

I don't think you need to add a redundant check for that, but perhaps an extra comment here clarifying would be a good idea.

Reply:
Yeh, I'm not sure why this was treated as a null return value rather than an error but the documentation for that belongs in the authentication service, not the route handler, so going to add a comment there instead.